### PR TITLE
Add services to sidebar

### PR DESCRIPTION
**Pull request by shuliakleonid**
add user name title
add services for Http requiest

### CODE REVIEW

File: exadel-trainee-app/package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "test": "ng test",
     "lint": "eslint \"**/*.{ts,tsx}\"",
     "lint:fix": "eslint \"**/*.{ts,tsx}\" --fix --color",
-    "lint:styles": "stylelint \"src/**/*.scss\" --fix --color --formatter"
+    "lint:styles": "stylelint \"src/**/*.scss\" --fix"
```
Comment (shuliakleonid) [inline exadel-trainee-app/package.json:12]: fix stylelint error when start pre-commit

File: exadel-trainee-app/package.json [hunk 2]
```diff
@@ -70,7 +70,8 @@
   },
   "lint-staged": {
     "*.{js,ts}": [
-      "npm run lint:fix && lint:styles",
+      "npm run lint:fix",
+      "npm run lint:styles",
```
Comment (shuliakleonid) [inline exadel-trainee-app/package.json:74]: fix husly pre-commit

File: exadel-trainee-app/src/interfaces/user.interfaces.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export interface UserResponseType {
+  id: number;
+  message: null;
+  isAuthenticated: boolean;
+  firstName: string;
+  lastName: string;
+  email: string;
+  roles: string[];
```
Comment (Inunor) [inline exadel-trainee-app/src/interfaces/user.interfaces.ts:8]: I think the role should be an array of Roles interfaces or something, not just string array

File: exadel-trainee-app/src/constants/mock-user-data.ts [hunk 4]
```diff
@@ -32,22 +32,49 @@ export const usersRoute: UserRoutesType = {
...
   {
-    value: Role.admin,
-    viewValue: UserViewValue.admin,
+    id: 3,
+    message: null,
+    isAuthenticated: true,
+    firstName: 'Coach',
+    lastName: 'Coachov',
+    email: 'ElevelAdministrator@gmail.com',
+    roles: ['coach'],
```
Comment (Inunor) [inline exadel-trainee-app/src/constants/mock-user-data.ts:76]: Same here about roles, could you please tell me why did you decide to make it just array of strings?

File: exadel-trainee-app/src/app/services/user-role.service.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+  constructor(private readonly userHttpService: UserHttpService) {}
+
+  private set activeUser(user: UserResponseType) {
+    this.userSubject$.next(user);
+  }
+
+  getActiveUser(): void {
+    this.userHttpService.getUser().subscribe((user) => {
+      this.activeUser = user;
```
Comment (Inunor) [inline exadel-trainee-app/src/app/services/user-role.service.ts:22]: It's ok, but in ngrx (redux) world and in javascript world in general it's a good practice to copy objects, not just copy link.`this.activeUser = { ...user }`

Comment (Inunor) [reply exadel-trainee-app/src/app/services/user-role.service.ts:22]: Everything else on services is done well from your side

File: exadel-trainee-app/src/app/services/user-role.service.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+import { UserResponseType } from '../../interfaces/user.interfaces';
+import { UserHttpService } from './user-role-http.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class UserRoleService {
```
Comment (Inunor) [inline exadel-trainee-app/src/app/services/user-role.service.ts:9]: I would suggest having a common name for these store services across app. You will have a bunch of store services, so something like [Feature]StoreService will be enough (UsersStoreService, RoleStoreService)

**Review by Inunor — CHANGES_REQUESTED** [review]

File: exadel-trainee-app/src/constants/mock-user-data.ts (hunk 4 above)
Comment (shuliakleonid) [reply exadel-trainee-app/src/constants/mock-user-data.ts:76]: It`s answer for request API

File: exadel-trainee-app/src/app/services/user-role.service.ts (hunk 5 above)
Comment (shuliakleonid) [reply exadel-trainee-app/src/app/services/user-role.service.ts:22]: Done, https://github.com/Inunor/internship/pull/73

File: exadel-trainee-app/src/app/services/user-role.service.ts (hunk 6 above)
Comment (shuliakleonid) [reply exadel-trainee-app/src/app/services/user-role.service.ts:9]: Done, like that 
image
https://github.com/Inunor/internship/pull/73

File: exadel-trainee-app/src/interfaces/user.interfaces.ts (hunk 3 above)
Comment (shuliakleonid) [reply exadel-trainee-app/src/interfaces/user.interfaces.ts:8]: I ask the backend developer for the structure of  request and answer is :
image

**Review by Inunor — APPROVED** [review]

**Pull request merged**